### PR TITLE
Logging polish (part 2)

### PR DESCRIPTION
**Pull request by tjvezina**
1. Omit line number from log context when zero
2. Corrected outdated documentation comment

### What is the current behaviour?
Logs from managed code have line number zero in the context, ex: `(UnityLogListener:0|.ctor) message`

### What is the new behaviour?
Logs from managed code now omit the line number, ex: `(UnityLogListener|.ctor) message`

Note: The same functionality does not apply to `FileLogHandler` as it allows the user to define a log context format string, making it confusing if part of that format is silently omitted in some cases. It would be nearly impossible to cleanly omit the line number in custom format strings anyway.

### What PRs does this depend on?
 - None

### Does this introduce a breaking change?
 - [ ] Yes 
 - [X] No

### Related
- https://github.com/decline-cookies/anvil-unity-core/pull/82

### CODE REVIEW

**Review by jkeon — DISMISSED** [review]
Looks good. 

Why would the line number be 0 ever though? Maybe I'm not fully understanding the case when it is 0?

**Comment by tjvezina** [comment]
It's just the default value returned by `StackFrame.GetFileLineNumber()` - so if the log context fails to resolve you'd see `(unknown|unknown)`, or more likely if the log comes from a DLL or other managed code, there is no file/line to point to, so we fall back on showing `DeclaringType.Name` and omit the line number, ex `(UnityLogListener|.ctor)`.

File: Logging/.CSProject/LogHandler/ConsoleLogHandler.cs [hunk 1]
```diff
@@ -19,7 +19,11 @@ public class ConsoleLogHandler : ILogHandler
             string callerName,
             int callerLine)
         {
-            message = $"({Path.GetFileNameWithoutExtension(callerPath)}:{callerLine}|{callerName}) {message}";
+            string callerFile = Path.GetFileNameWithoutExtension(callerPath);
+
+            string context = (callerLine > 0 ? $"{callerFile}:{callerLine}|{callerName}" : $"{callerFile}|{callerName}");
+
+            message = $"({context}) {message}";
```
Comment (mbaker3) [inline Logging/.CSProject/LogHandler/ConsoleLogHandler.cs:26]: Same as comment here. Remove extra string allocation
https://github.com/decline-cookies/anvil-unity-core/pull/82#discussion_r990223627

**Review by mbaker3 — CHANGES_REQUESTED** [review]
> Note: The same functionality does not apply to FileLogHandler as it allows the user to define a log context format string, making it confusing if part of that format is silently omitted in some cases. It would be nearly impossible to cleanly omit the line number in custom format strings anyway.

Yea that's a shame because it's the direction all of the handlers will move eventually.
When we do take on that effort maybe we just include two different versions of the caller line token.
`LOG_CONTEXT_CALLER_LINE_ALWAYS` and `LOG_CONTEXT_CALLER_LINE` where one would only show conditionally, maybe with one way to associate the separator with the conditional version so they could both be omitted effectively when unknown.

I suppose the other option is just to replace line number with `?` or `NaN` or something when it's not known.

**Review by mbaker3 — DISMISSED** [review]

File: Logging/.CSProject/LogHandler/ConsoleLogHandler.cs [hunk 2]
```diff
@@ -19,7 +19,16 @@ public class ConsoleLogHandler : ILogHandler
...
+            string callerFile = Path.GetFileNameWithoutExtension(callerPath);
+
+            if (callerLine > 0)
+            {
+               message = $"({callerFile}:{callerLine}|{callerName}) {message}"
+            }
+            else
+            {
+               message = $"({callerFile}|{callerName}) {message}"
+            }
```
Comment (mbaker3) [inline Logging/.CSProject/LogHandler/ConsoleLogHandler.cs:31]:
```suggestion
            if (callerLine > 0)
            {
               message = $"({callerFile}:{callerLine}|{callerName}) {message}";
            }
            else
            {
               message = $"({callerFile}|{callerName}) {message}";
            }
```

**Review by mbaker3 — APPROVED** [review]

**Comment by jkeon** [comment]
Ahh gotcha. Ok cool thanks

**Pull request merged**